powerauth-rest-client-spring/src/main/java/com/wultra/security/powerauth/rest/client/PowerAuthFido2RestClient.java

```diff
@@ -0,0 +1,258 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package com.wultra.security.powerauth.rest.client;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.wultra.core.rest.client.base.DefaultRestClient;
+import com.wultra.core.rest.client.base.RestClient;
+import com.wultra.core.rest.client.base.RestClientException;
+import com.wultra.security.powerauth.client.PowerAuthFido2Client;
+import com.wultra.security.powerauth.client.model.entity.fido2.AuthenticatorAssertionResponse;
+import com.wultra.security.powerauth.client.model.entity.fido2.AuthenticatorParameters;
+import com.wultra.security.powerauth.client.model.error.PowerAuthClientException;
+import com.wultra.security.powerauth.client.model.error.PowerAuthError;
+import com.wultra.security.powerauth.client.model.request.fido2.*;
+import com.wultra.security.powerauth.client.model.response.InitActivationResponse;
+import com.wultra.security.powerauth.client.model.response.fido2.*;
+import io.getlime.core.rest.model.base.request.ObjectRequest;
+import io.getlime.core.rest.model.base.response.ObjectResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Class implementing a PowerAuth REST client.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ *
+ */
+public class PowerAuthFido2RestClient implements PowerAuthFido2Client {
+
+    private static final Logger logger = LoggerFactory.getLogger(PowerAuthFido2RestClient.class);
+
+    private static final String PA_REST_FIDO2_PREFIX = "/fido2";
+    private static final MultiValueMap<String, String> EMPTY_MULTI_MAP = new LinkedMultiValueMap<>();
+
+    private final RestClient restClient;
+    private final ObjectMapper objectMapper = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+    /**
+     * PowerAuth REST client constructor.
+     *
+     * @param baseUrl BASE URL of REST endpoints.
+     */
+    public PowerAuthFido2RestClient(String baseUrl) throws PowerAuthClientException {
+        this(baseUrl, new PowerAuthRestClientConfiguration());
+    }
+
+    /**
+     * PowerAuth REST client constructor.
+     *
+     * @param baseUrl Base URL of REST endpoints.
+     */
+    public PowerAuthFido2RestClient(String baseUrl, PowerAuthRestClientConfiguration config) throws PowerAuthClientException {
+        final DefaultRestClient.Builder builder = DefaultRestClient.builder().baseUrl(baseUrl)
+                .acceptInvalidCertificate(config.getAcceptInvalidSslCertificate())
+                .connectionTimeout(config.getConnectTimeout())
```

Comment:
Please include also `HttpMaxIdleTime` configuration.

Reply:
Let's fix this separately in a separate ticket, because this change is also related to PowerAuthClient.